src/app/newSettings/containers/InvoicesContainer.tsx

```diff
@@ -1,31 +1,39 @@
 import { Invoice } from '@internxt/sdk/dist/drive/payments/types';
 import { useTranslationContext } from '../../i18n/provider/TranslationProvider';
-import { useEffect, useState } from 'react';
 import Section from '../Sections/General/components/Section';
-import paymentService from '../../payment/services/payment.service';
 import Card from '../../shared/components/Card';
 import InvoicesList from '../components/Invoices/InvoicesList';
+import { Spinner } from '@internxt/internxtui';
 
-const Invoices = ({ className = '', subscriptionId }: { className?: string; subscriptionId: string }): JSX.Element => {
+const Invoices = ({
+  className = '',
+  invoices,
+  state,
+}: {
+  className?: string;
+  invoices: Invoice[];
+  state: 'ready' | 'loading' | 'empty';
+}): JSX.Element => {
   const { translate } = useTranslationContext();
-  const [state, setState] = useState<{ tag: 'ready'; invoices: Invoice[] } | { tag: 'loading' | 'empty' }>({
-    tag: 'loading',
-  });
-  const isEmpty = state.tag === 'empty';
 
-  useEffect(() => {
-    paymentService
-      .getInvoices({ subscriptionId })
-      .then((invoices) => setState({ tag: 'ready', invoices }))
-      .catch(() => setState({ tag: 'empty' }));
-  }, []);
-
-  const invoices = state.tag === 'ready' ? state.invoices : [];
+  const isEmpty = state === 'empty';
 
   return (
     <Section className={className} title={translate('views.account.tabs.billing.invoices.head')}>
-      <Card className={`${isEmpty ? 'h-40' : 'pb-0'}`}>
-        <InvoicesList invoices={invoices} state={state.tag} />
+      <Card className={`${isEmpty && 'h-20'}`}>
+        {state === 'ready' ? (
+          <InvoicesList invoices={invoices} state={state} />
+        ) : state === 'loading' ? (
+          <div className="flex h-10 items-center justify-center">
+            <Spinner className="h-5 w-5" />
+          </div>
+        ) : state === 'empty' ? (
+          <div className="flex h-full items-center justify-center text-center">
+            <p className="font-regular text-base text-gray-60">
+              {translate('views.account.tabs.billing.invoices.empty')}
+            </p>
+          </div>
+        ) : null}
```

Comment:
What do you think of this way of showing the different components depending on the state? Perhaps it is clearer than a ternary conditional mode.

```
const stateViews = {
  ready: <InvoicesList invoices={invoices} state={state} />,
  loading: (
    <div className="flex h-10 items-center justify-center">
      <Spinner className="h-5 w-5" />
    </div>
  ),
  empty: (
    <div className="flex h-full items-center justify-center text-center">
      <p className="font-regular text-base text-gray-60">
        {translate('views.account.tabs.billing.invoices.empty')}
      </p>
    </div>
  ),
};
```

```suggestion
  {stateViews[state] || null}
```